lib/plugins/base.py

```diff
@@ -0,0 +1,66 @@
+from html.parser import HTMLParser
+
+from markdown import markdown
+
+
+# inspired by:
+# https://stackoverflow.com/questions/14694482/
+class HTMLFilter(HTMLParser):
+    text = ""
+    href = ""
+    li_count = 0
+    inside_blockquote = False
+    last_start_tag = ""
+
+    def handle_starttag(self, tag, attrs):
+        if tag == "blockquote":
+            self.inside_blockquote = True
+        elif tag == "ol":
+            if self.inside_blockquote:
+                self.text += ">"
+            self.li_count = 1
+        elif tag == "ul":
+            if self.inside_blockquote:
+                self.text += ">"
+            self.li_count = 0
+        elif tag == "li":
+            if self.inside_blockquote:
+                self.text += "> "
+            if self.li_count:
+                self.text += f"{self.li_count}. "
+                self.li_count += 1
+            else:
+                self.text += "- "
+        elif tag == "p" or (tag[0]=="h" and tag[1].isdigit()):
+            # add an empty line before paragraphs and headings
+            # do blockquoting as needed
+            if self.inside_blockquote:
+                if self.last_start_tag == "blockquote":
+                    p_sep = "\n\n> "
+                else:
+                    p_sep = "\n>\n> "
+            else:
+                p_sep = "\n\n"
+            self.text = self.text.rstrip("\n") + p_sep
```

Comment:
adding extra `\n` will also cause problem in situation like if the input contains something like this:
```markdown
- a

- b
```
the output will be this:
```

-  

a
- 

b


```

Reply:
I suppose you've tested this?

Reply:
handling this now by turning line breaks on and off depending on context.